src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Options/Altinn2Provider/Altinn2MetadataApiClient.cs

```diff
@@ -0,0 +1,22 @@
+#nullable enable
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Altinn.App.PlatformServices.Options.Altinn2Provider
+{
+    public class Altinn2MetadataApiClient
+    {
+        private readonly HttpClient _client;
+        public Altinn2MetadataApiClient(HttpClient client)
+        {
+            _client = client;
+        }
+        public async Task<MetadataCodelistResponse> GetAltinn2Codelist(string id, string langCode, int? version = null)
+        {
+            var response = await _client.GetAsync($"https://www.altinn.no/api/metadata/codelists/{id}/{version?.ToString()??string.Empty}?language={langCode}");
```

Comment:
@SandGrainOne is it a problem if we always use the production environment for Altinn 2 for these data? Or do we need environment support?

Reply:
I don't think environment support is important, but it is given for free through app settings configuration files. I see no reason not use it.